libs/audit-service/src/main/java/uk/gov/di/ipv/core/library/service/AuditService.java

```diff
@@ -2,46 +2,90 @@
 
 import com.fasterxml.jackson.core.JsonProcessingException;
 import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 import software.amazon.awssdk.http.urlconnection.UrlConnectionHttpClient;
+import software.amazon.awssdk.services.sqs.SqsAsyncClient;
 import software.amazon.awssdk.services.sqs.SqsClient;
 import software.amazon.awssdk.services.sqs.model.SendMessageRequest;
+import software.amazon.awssdk.services.sqs.model.SendMessageResponse;
 import uk.gov.di.ipv.core.library.auditing.AuditEvent;
+import uk.gov.di.ipv.core.library.exception.AuditException;
 import uk.gov.di.ipv.core.library.exceptions.SqsException;
+import uk.gov.di.ipv.core.library.helpers.LogHelper;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
 
 import static software.amazon.awssdk.regions.Region.EU_WEST_2;
+import static uk.gov.di.ipv.core.library.config.CoreFeatureFlag.SQS_ASYNC;
 import static uk.gov.di.ipv.core.library.config.EnvironmentVariable.SQS_AUDIT_EVENT_QUEUE_URL;
 
 public class AuditService {
-    private final SqsClient sqs;
-    private final String queueUrl;
+    private static final Logger LOGGER = LogManager.getLogger();
+    private final SqsClients sqsClients;
     private final ObjectMapper objectMapper;
+    private final ConfigService configService;
+    private List<CompletableFuture<SendMessageResponse>> events = new ArrayList<>();
 
-    public AuditService(SqsClient sqs, ConfigService configService) {
-        this.sqs = sqs;
-        this.queueUrl = configService.getEnvironmentVariable(SQS_AUDIT_EVENT_QUEUE_URL);
+    public AuditService(SqsClients sqsClients, ConfigService configService) {
+        this.sqsClients = sqsClients;
+        this.configService = configService;
         this.objectMapper = new ObjectMapper();
     }
 
-    public AuditService(SqsClient sqs, ConfigService configService, ObjectMapper objectMapper) {
-        this.sqs = sqs;
-        this.queueUrl = configService.getEnvironmentVariable(SQS_AUDIT_EVENT_QUEUE_URL);
+    public AuditService(
+            SqsClients sqsClients, ConfigService configService, ObjectMapper objectMapper) {
+        this.sqsClients = sqsClients;
+        this.configService = configService;
         this.objectMapper = objectMapper;
     }
 
-    public static SqsClient getSqsClient() {
-        return SqsClient.builder()
-                .region(EU_WEST_2)
-                .httpClientBuilder(UrlConnectionHttpClient.builder())
-                .build();
+    public static SqsClients getSqsClients() {
+        return new SqsClients(
+                SqsClient.builder()
+                        .region(EU_WEST_2)
+                        .httpClientBuilder(UrlConnectionHttpClient.builder())
+                        .build(),
+                SqsAsyncClient.builder().region(EU_WEST_2).build());
```

Comment:
Deliberately using the deafult async client (netty), as we've removed use of the crt client. We should switch if/when we bring the crt client back.

Reply:
It might be worth trying the async CRT client, in case it's more reliable - the Netty client has a slower startup time which might wipe out any gains from the async calls..!

Reply:
I can't reproduce the segfault with the async client so I'd like to try it in lower environments and see...